app/models/course/assessment/submission.rb

```diff
@@ -92,6 +92,25 @@ def auto_grade!
     AutoGradingJob.perform_later(self)
   end
 
+  # Return the next unanswered question.
+  #
+  # @return [Course::Assessment::Question|nil] the next unanswered question or nil if all
+  #   questions have been correctly answered.
+  def next_unanswered_question
+    all_questions = assessment.questions
+    correctly_answered_questions =
+      all_questions.where(id: answers.where(correct: true).select(:question_id))
```

Comment:
I see some problems now:

course/assessment/questions_concern has:
 - correctly_answered_questions
 - not_correctly_answered

and course/assessment/submission.rb has
 - next_unanswered_question which looks like assessment.questions#step